README.md

```diff
@@ -137,36 +137,33 @@ const UserSchema = new Schema({
   }
 });
 
-// NOTE: `this: UserDocument` and `this: UserModel` is to tell TS the type of `this' value using the "fake this" feature
+// NOTE: `this: UserDocument` and `this: UserModel` is required for virtual properties to tell TS the type of `this' value using the "fake this" feature
 // you will need to add these in after your first ever run of the CLI
-
 UserSchema.virtual("name").get(function (this: UserDocument) {
   return `${this.firstName} ${this.lastName}`;
 });
 
-// method functions
+// method functions, cast to UserStatics
 UserSchema.methods = {
-  isMetadataString(this: UserDocument) {
+  isMetadataString() {
     return typeof this.metadata === "string";
   }
-};
+} as UserMethods;
```

Comment:
Why was this changed to a cast?
I don't see any advantages (but that could be me), and it reduces type-safety.

Reply:
Casting enforces the function types in this case. If you take a look at the mongoose typings for these schema properties (methods, statics, and query) they look like this:

```
/** Object of currently defined methods on this schema. */
    methods: {
      [F in keyof T]: T[F]
    };
    /** Object of currently defined statics on this schema. */
    statics: any;
    /** Object of currently defined query helpers on this schema. */
    query: any;
```

This casting doesn't reduce the type-safety, if anything it improves it. It can be seen as more of "type assertion" then typecasting. It also substantially reduces the changes a user needs to make to use the library (no need for `this: UserDocument`)